src/core/test/fuzzing/src/it/scala/DatabricksUtilities.scala

```diff
@@ -0,0 +1,260 @@
+         |  "notebook_task": {
+         |    "notebook_path": "$notebookPath",
+         |    "base_parameters": []
+         |  },
+         |  "libraries": $libraries
+         |}
+      """.stripMargin
+
+    databricksPost("jobs/runs/submit", body).select[Int]("run_id")
+  }
+
+  private def getRunStatuses(runId: Int): (String, Option[String]) = {
+    val runObj = databricksGet(s"jobs/runs/get?run_id=$runId")
+    val stateObj = runObj.select[JsObject]("state")
+    val lifeCycleState = stateObj.select[String]("life_cycle_state")
+    if (lifeCycleState == "TERMINATED") {
+      val resultState = stateObj.select[String]("result_state")
+      (lifeCycleState, Some(resultState))
+    } else {
+      (lifeCycleState, None)
+    }
+  }
+
+  def getRunUrlAndNBName(runId: Int): (String, String) = {
+    val runObj = databricksGet(s"jobs/runs/get?run_id=$runId").asJsObject()
+    val url = runObj.select[String]("run_page_url")
+      .replaceAll("westus", region) //TODO this seems like an ADB bug
+    val nbName = runObj.select[String]("task.notebook_task.notebook_path")
+    (url, nbName)
+  }
+
+  def monitorJob(runId: Integer,
+                 interval: Int = 2000,
+                 timeout: Int = 10000,
+                 logLevel: Int = 1): Future[Unit] = {
+    Future {
+      var finalState: Option[String] = None
+      var lifeCycleState: String = "Not Started"
+      val startTime = System.currentTimeMillis()
+      val (url, nbName) = getRunUrlAndNBName(runId)
+      if (logLevel >= 1) println(s"Started Monitoring notebook $nbName, url: $url")
+
+      while (finalState.isEmpty &
+        (System.currentTimeMillis() - startTime) < timeout &
+        lifeCycleState != "INTERNAL_ERROR"
+      ) {
+        val (lcs, fs) = getRunStatuses(runId)
+        finalState = fs
+        lifeCycleState = lcs
+        if (logLevel >= 2) println(s"Job $runId state: $lifeCycleState")
+        blocking {
+          Thread.sleep(interval.toLong)
+        }
+      }
+
+      val error = finalState match {
+        case Some("SUCCESS") =>
+          if (logLevel >= 1) println(s"Notebook $nbName Suceeded")
+          None
+        case Some(state) =>
+          Some(new RuntimeException(s"Notebook $nbName failed with state $state. " +
+            s"For more information check the run page: \n$url\n"))
+        case None if lifeCycleState == "INTERNAL_ERROR" =>
+          Some(new RuntimeException(s"Notebook $nbName failed with state $lifeCycleState. " +
+            s"For more information check the run page: \n$url\n"))
+        case None =>
+          Some(new TimeoutException(s"Notebook $nbName timed out after $timeout ms," +
+            s" job in state $lifeCycleState, " +
+            s" For more information check the run page: \n$url\n "))
+      }
+
+      error.foreach { error =>
+        if (logLevel >= 1) print(error.getMessage)
+        throw error
+      }
+
+    }(ExecutionContext.global)
```

Comment:
Being standard doesn't make it better.

    Foo({
      ... 50 lines of stuff ...
    }, bar())

is hard to read since `bar` is related to the *distant* `Foo`.  OTOH,

    def stuff = {
      ... 50 lines of stuff ...
    }
    Foo(stuff, bar())

is nice and readable.